include/arch/arc/v2/irq.h

```diff
@@ -68,6 +69,82 @@ extern void z_irq_spurious(void *unused);
 })
 
 
+/**
+ * Configure a 'direct' static interrupt.
+ *
+ * When firq has no separate stack(CONFIG_ARC_FIRQ_STACK=N), it's not safe
+ * to call C ISR handlers because sp will be switched to bank1's sp which
+ * is undefined value.
+ * So for this case, the priority cannot be set to 0 but next level 1
+ *
+ * When firq has separate stack (CONFIG_ARC_FIRQ_STACK=y) but at the same
+ * time stack checking is enabled (CONFIG_ARC_STACK_CHECKING=y)
+ * the stack checking can raise stack check exception as sp is switched to
+ * firq's stack (bank1's sp). So for this case, the priority cannot be set
+ * to 0 but next level 1.
+ *
+ * Note that for the above cases, if application still wants to use firq by
+ * setting priority to 0. Application can call z_irq_priority_set again.
+ * Then it's left to application to handle the details of firq
+ *
+ * See include/irq.h for details.
+ * All arguments must be computable at build time.
+ */
+#define Z_ARCH_IRQ_DIRECT_CONNECT(irq_p, priority_p, isr_p, flags_p) \
+({ \
+	Z_ISR_DECLARE(irq_p, ISR_FLAG_DIRECT, isr_p, NULL); \
+	if (IS_ENABLED(CONFIG_ARC_FIRQ) && (priority_p == 0)) { \
+		if (!IS_ENABLED(CONFIG_ARC_FIRQ_STACK) || \
+		(IS_ENABLED(CONFIG_ARC_FIRQ_STACK) && \
+		IS_ENABLED(CONFIG_ARC_STACK_CHECKING))) { \
```

Comment:
Remove this if statement, both clauses are the same

Reply:
fixed